src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs

```diff
@@ -31,83 +31,59 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
 
         void CreateQueueIfNecessary(string address, string identity)
         {
-            if (address == null)
+            Guard.AgainstNullAndEmpty(nameof(address), address);
+
+            var msmqAddress = MsmqAddress.Parse(address);
+            
+            Logger.Debug($"Trying to open queue '{address}'.");
+
+            MessageQueue queue;
+            if (!MsmqUtilities.TryOpenQueue(msmqAddress, out queue))
             {
-                return;
+                Logger.Warn($"Queue '{address}' does not exist.");
+                Logger.Debug($"Creating queue: {address}");
```

Comment:
This is vierd, how about changing the WARN to "Queue '{address}' does not exist. Going to create it" or similar and remove the .Debug?